File: source/repos/BlackJack/BlackJack.ViewModels/GameViews/ContinueGameView.cs
﻿using BlackJack.DataAccess.Enums;
using System.Collections.Generic;

namespace BlackJack.ViewModels.GameViews
{
    public class ContinueGameView
    {
        public StatusType Status { get; set; }
        public string Winner { get; set; }
        public PlayerContinueGameView Player { get; set; }
        public List<BotContinueGameViewItem> Bots { get; set; }

        public ContinueGameView()
        {
            Bots = new List<BotContinueGameViewItem>();
            Player = new PlayerContinueGameView();
        }
    }
    public class BotContinueGameViewItem
    {
        public string Name { get; set; }
        public List<CardContinueGameViewItem> Cards { get; set; }
    }
    public class PlayerContinueGameView
    {
        public string Name { get; set; }
        public List<CardContinueGameViewItem> Cards { get; set; }
    }
    public class CardContinueGameViewItem
    {
        public CardRankType Rank { get; set; }
        public CardSuitType Suit { get; set; }
    }
}

File: source/repos/BlackJack/BlackJack.WEB/ClientApp/src/app/shared/entities/game/continue-game.view..ts
import { CardRankType } from 'src/app/shared/enums/card-rank-type.view';
import { CardSuitType } from 'src/app/shared/enums/card-suit-type.view';
import { StatusType } from 'src/app/shared/enums/status-type.enum.view';

export class ContinueGameView {
    status: StatusType;
    winner: string;
    player:PlayerContinueGameView;
    bots:BotContinueGameViewItem[] = [];
}
export class PlayerContinueGameView{
    name:string;
    cards:CardContinueGameViewItem[] = [];
}
export class BotContinueGameViewItem{
    name:string;
    cards:CardContinueGameViewItem[] = [];
}
export class CardContinueGameViewItem{
    rank:CardRankType;
    suit:CardSuitType;
}